Ginger/GingerCoreNET/ActionsLib/ActionConversionUtils.cs

```diff
@@ -0,0 +1,182 @@
+                            {
+                                // get the index of the action that is being converted 
+                                int selectedActIndex = currentActivity.Acts.IndexOf(act);
+
+                                // convert the old action
+                                Act newAct = ((IObsoleteAction)act).GetNewAction();
+                                if (convertToPOMAction && newAct.GetType().Name == ActUIElementClassName)
+                                {
+                                    newAct = GetMappedElementFromPOMForAction(newAct, selectedPOMObjectName);
+                                }
+                                currentActivity.Acts.Insert(selectedActIndex + 1, newAct);
+
+                                // set obsolete action in the activity as inactive
+                                act.Active = false;
+                            }
+                        }
+
+                        // if the user has not chosen any target application in the combobox then, we set it as empty
+                        if (isDefaultTargetApp && !string.IsNullOrEmpty(strTargetApp))
+                        {
+                            currentActivity.TargetApplication = strTargetApp;
+                        }
+                        else
+                        {
+                            currentActivity.TargetApplication = string.Empty;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+            }
+        }
+        
+        /// <summary>
+        /// This method is used to find the relative element from POM for the existing action
+        /// </summary>
+        /// <param name="newActUIElement"></param>
+        /// <param name="pomModelObject"></param>
+        /// <returns></returns>
+        public Act GetMappedElementFromPOMForAction(Act newActUIElement, string pomModelObject)
+        {
+            ObservableList<ApplicationPOMModel> pomLst = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ApplicationPOMModel>();
+            ApplicationPOMModel selectedPOM = pomLst.Where(x => x.Guid.ToString() == pomModelObject).SingleOrDefault();
+
+            ElementInfo elementInfo = null;
+            string locateValue = Convert.ToString(newActUIElement.GetType().GetProperty(ActUIElementLocateValueField).GetValue(newActUIElement, null));
+            eLocateBy elementLocateBy = GeteLocateByEnumItem(Convert.ToString(newActUIElement.GetType().GetProperty(ActUIElementElementLocateByField).GetValue(newActUIElement, null)));
+
+            var matchingExistingLocator = selectedPOM.MappedUIElements.Where(x => x.Locators.Any(y => y.LocateBy == elementLocateBy && y.LocateValue == locateValue));
```

Comment:
for "y.LocateValue == locateValue"
please compare the strings after both sides done Trim() and ToLower() or better use String.Compare 
that to avoid missing match only because of diffrent case or some space before or after

Reply:
Done